lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -486,10 +486,12 @@ def add_reference(post, authorization, options = {})
         post[:originalReference] = original_reference
       end
 
-      def add_mpi_data_for_network_tokenization_card(post, payment)
+      def add_mpi_data_for_network_tokenization_card(post, payment, options)
+        return if options[:skip_mpi_data] == 'Y'
+
         post[:mpiData] = {}
         post[:mpiData][:authenticationResponse] = 'Y'
-        post[:mpiData][:cavv] = payment.payment_cryptogram
+        post[:mpiData][:cavv] = payment.payment_cryptogram unless options[:skip_mpi_data] == 'Y'
```

Comment:
I think this `unless` statement is irrelevant, right? If we ever reach it, it should evaluate false. If it's going to evaluate true, we'll never reach it because we'll `return` at line 490

Reply:
Oh, good catch! I meant to take that out.